src/provider/sequencer.ts

```diff
@@ -106,6 +111,16 @@ export class SequencerProvider implements ProviderInterface {
     return StarknetChainId.TESTNET;
   }
 
+  private parseUrlOrPath(baseUrl: string, defaultPath: string, urlOrPath?: string) {
+    if (urlOrPath) {
+      if (isUrl(urlOrPath)) {
+        return urlOrPath;
+      }
+      return urljoin(baseUrl, urlOrPath);
+    }
+    return urljoin(baseUrl, defaultPath);
+  }
+
```

Comment:
I think it can be done simply 
```return isUrl(urlOrPath) ? urlOrPath : urljoin(baseUrl, urlOrPath ?? defaultPath)```

My opinion is to name functions by the result. Like createUrl() or getUrl because it returns full URI from optional path/url and does not return a list of parsed elements.

Reply:
I think this could be extracted to utils with isUrl as it is generalized, and doesn't use any private props from class